File: case01/java/Cliente.java
public class Cliente
{
    private String nome, email, senha;
    private char sexo;
    private int idade;
    private Livro livro;
    
    public Cliente(String nome, int idade, char sexo, String email, String senha, Livro livro){
        this.nome = nome;
        this.idade = idade;
        this.sexo = sexo;
        this.email = email;
        this.senha = senha;
        this.livro = livro;
    }
    
    public Cliente(String nome, int idade, String email, String senha, Livro livro){
        this.nome = nome;
        this.idade = idade;
        this.email = email;
        this.senha = senha;
        this.livro = livro;
    }
    
    public Cliente(String nome, char sexo, String email, String senha, Livro livro){
        this.nome = nome;
        this.sexo = sexo;
        this.email = email;
        this.senha = senha;
        this.livro = livro;
    }
    
    public String getNome(){
        return nome;
    }
    public int getIdade(){
        return idade;
    }
    public char getSexo(){
        return sexo;
    }
    public String getEmail(){
        return email;
    }
    public String getSenha(){
        return senha;
    }
    public Livro getLivro(){
        return livro;
    }
    public void setNome(String nome){
        this.nome = nome;
    }
    public void setIdade(int idade){
        this.idade = idade;
    }
    public void setSexo(char sexo){
        this.sexo = sexo;
    }
    public void setEmail(String email){
        this.email = email;
    }
     public void setSenha(String senha){
        this.senha = senha;
    }
    public void setLivro(Livro livro){
        this.livro = livro;
    }
    public void imprimeInformacoes(){
         System.out.println("Nome: " +nome+
         ", Idade: " +idade+
         ", Sexo: " +sexo+
         ", \nEmail: " +email+
         ", Senha: " +senha+
         ", Livro: " +livro);
    }
    public String toString(){
        return "O cliente " +nome+ " está como livro " +livro.getTitulo()+ 
        ", \ne o preço por página deste livro é R$ " +livro.calcularPrecoPorPagina();
    }
}

File: case01/Cliente.py
from Livro import Livro

class Cliente:
    def __init__(self, nome: str, email: str, senha: str, livro: Livro, sexo = None, idade = None):
        if sexo is not None and idade is not None:
            self.nome = nome
            self.idade = idade
            self.sexo = sexo
            self.email = email
            self.senha = senha
            self.livro = livro
        elif sexo is None and idade is not None:
            self.nome = nome
            self.idade = idade
            self.email = email
            self.senha = senha
            self.livro = livro
        else:
            self.nome = nome
            self.sexo = sexo
            self.email = email
            self.senha = senha
            self.livro = livro

    def get_nome(self):
        return self.nome
    
    def get_idade(self):
        return self.idade
    
    def get_sexo(self):
        return self.sexo
    
    def get_email(self):
        return self.email
    
    def get_senha(self):
        return self.senha
    
    def get_livro(self):
        return self.livro
    
    def set_nome(self, nome):
        self.nome = nome

    def set_idade(self, idade):
        self.idade = idade

    def set_sexo(self, sexo):
        self.sexo = sexo
    
    def set_email(self, email):
        self.email = email

    def set_senha(self, senha):
        self.senha = senha

    def set_livro(self, livro):
        self.livro = livro
        
    def __str__(self):
        return f"Nome: {self.nome}, está com o livro: {self.livro.get_titulo()} \nO preço por página deste livro é R$ {self.livro.calcularPrecoPorPagina()}" 

